packages/next/src/build/webpack-config.ts

```diff
@@ -1198,12 +1198,14 @@ export default async function getBaseWebpackConfig(
                   chunks: 'all' as const,
                   name: 'framework',
                   layer: RSPACK_DEFAULT_LAYERS_REGEX,
-                  test: /[/]node_modules[/](react|react-dom|next[/]dist[/]compiled[/](react|react-dom)(-experimental)?)[/]/,
+                  test: new RegExp(
+                    `^(${topLevelFrameworkPaths.map((p) => `(${p.replace(/[.*+?^${}()|[\]\\]/g, '\\$&')})`).join('|')})`
```

Comment:
Is it possible for this to use https://github.com/vercel/next.js/blob/canary/packages/next/src/shared/lib/escape-regexp.ts ?

Reply:
ok, i have used it.